### PR TITLE
Update the proposal flow to indicate GH should be checked first

### PR DESCRIPTION
**Pull request by iwiznia**
No test, no QA, just a contributing guidelines change.

cc @Expensify/contributor-management @sidferreira

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -59,9 +59,9 @@ In this scenario, it’s possible that you found a bug or enhancement that we ha
 ## Working on Expensify Jobs
 *Reminder: For technical guidance please refer to the [README](https://github.com/Expensify/App/blob/main/README.md)*.
 
-#### Express interest for the job on Upwork.com
+#### Check GitHub for existing proposals from other users
 
-1. If you are interested in working on a job posted in Upwork, click **Submit a Proposal** in Upwork to express your interest to the Expensify team.
+1. If someone already posted a good solution, most probably they will be assigned the issue. You can still post proposals, but you should know that the best proposal that addresses the problem will be selected, all things equal we prioritize the first proposal.
```
Comment (michaelhaxhiu) [inline CONTRIBUTING.md:64]: I think we could make this a little bit more clear with re-phrasing? I find `most probably they will be assigned the issue` to be a little unclear still. What if we re-structure like this?

>Expensify reviews all solution proposals on a first come first serve basis. If you see other contributors have already proposed a solution, you can still provide a solution proposal and we will review it if we don't select the first-received proposal(s). We look for the _earliest provided_ and _best_ proposed solution that addresses the job.

Comment (michaelhaxhiu) [reply CONTRIBUTING.md:64]: Take what you like and leave what you don't like, just suggesting this for more clarity really.

Comment (iwiznia) [reply CONTRIBUTING.md]: ok took most of it. Check it out

Comment (michaelhaxhiu) [reply CONTRIBUTING.md]: Looking

**Review by michaelhaxhiu — COMMENTED** [review]
Looks nice and clean!

File: CONTRIBUTING.md (hunk 1 above)
Comment (michaelhaxhiu) [reply CONTRIBUTING.md]: Looks great. Can I click [Resolve conversation] on this?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -59,9 +59,9 @@ In this scenario, it’s possible that you found a bug or enhancement that we ha
 ## Working on Expensify Jobs
 *Reminder: For technical guidance please refer to the [README](https://github.com/Expensify/App/blob/main/README.md)*.
 
-#### Express interest for the job on Upwork.com
+#### Check GitHub for existing proposals from other users
 
-1. If you are interested in working on a job posted in Upwork, click **Submit a Proposal** in Upwork to express your interest to the Expensify team.
+1. Expensify reviews all solution proposals on a first come first serve basis. If you see other contributors have already proposed a solution, you can still provide a solution proposal and we will review it. We look for the earliest provided and best proposed solution that addresses the job.
```
Comment (sidferreira) [inline CONTRIBUTING.md:64]: `earliest provided and best proposed solution that addresses the job`
vs
`earliest provided best proposed solution that addresses the job`

the 1st sounds like focusing in the `earliest`, the 2nd sounds like a bit better IMHO

@iwiznia

**Review by jasperhuangg — APPROVED** [review]

**Review by jasperhuangg — DISMISSED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (jasperhuangg) [reply CONTRIBUTING.md:64]: A comma would probably be good here:

`We look for the earliest provided, best proposed solution that addresses the job.`

Comment (sidferreira) [reply CONTRIBUTING.md:64]: @jasperhuangg lol self taught English, grammar is not my strong skill 😂

Comment (Beamanator) [reply CONTRIBUTING.md:64]: Agreed that this seems a bit better ^ here's the suggestion edit so we can commit it quickly :)
```suggestion
1. Expensify reviews all solution proposals on a first come first serve basis. If you see other contributors have already proposed a solution, you can still provide a solution proposal and we will review it. We look for the earliest provided, best proposed solution that addresses the job.
```

**Comment by iwiznia** [comment]
Updated with the suggestions

**Review by Beamanator — APPROVED** [review]

**Review by Jag96 — APPROVED** [review]

**Review by jasperhuangg — APPROVED** [review]

**Pull request merged**